### PR TITLE
Add Datadog integration instructions

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Datadog integration instructions

Datadog-Spice-ai-OSS-Docs (3)

### CODE REVIEW

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: "Datadog"
+sidebar_label: "Datadog"
+description: 'Monitoring Spice instances with Datadog'
+pagination_prev: 'clients/index'
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md:5]: Explicit ordering.

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+---
+title: "Datadog"
+sidebar_label: "Datadog"
+description: 'Monitoring Spice instances with Datadog'
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
description: 'Monitoring Spice with Datadog'
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+pagination_prev: 'clients/index'
+pagination_next: null
+---
+
+The Spice runtime metrics could be integrated to Datadog via Spice [Prometheus endpoint](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
Spice can be monitored with [Datadog](https://www.datadoghq.com/) using the [Prometheus metrics endpoint](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## Spice Prometheus endpoint configuration
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
## Metrics Endpoint Configuration
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+By default, Prometheus endpoint is running on port `9000`. Review Spice runtime output after it is started to confirm metrics endpoint is correctly displayed.
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
The metrics endpoint uses port `9000` by default. The metrics endpoint configuration is logged at startup.
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+```bash
+2024-07-15T21:48:00.158267Z  INFO spiced: Metrics listening on 127.0.0.1:9000
+```
+
+Use `--metrics` param to configure Prometheus endpoint, for example
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
Pass the `--metrics` parameter to bind to a specific port. For example, to bind to port `9090`:
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+or when using Docker:
+
+```Dockerfile
+FROM spiceai/spiceai:latest
+
+# Docker configuration ...
+
+# Run Promeheus metrics exporter on port 9090
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
# Configure the metrics endpoint on port 9090
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+CMD ["--metrics", "0.0.0.0:9090"]
+EXPOSE 9090
+```
+
+Prometheus endpoint could be verified by using HTTP GET request, for example:
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
Configuration of the metrics endpoint can be verified using a HTTP GET request, for example:
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+# TYPE datasets_count gauge
+datasets_count{engine="None"} 1
+datasets_count{engine="arrow"} 1
+...
+```
+
+## Datadog Agent configuration
+
+Prerequisite: [Datadog Agent version 6.5.0 or later is installed](https://docs.datadoghq.com/getting_started/agent/). Datadog Agent supports Prometheus metrics scraping by default, no additional packages are required on host or container.
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
Prerequisite: [Datadog Agent version 6.5.0 or later is installed](https://docs.datadoghq.com/getting_started/agent/). The Datadog Agent supports Prometheus metrics scraping by default, no additional packages are required on host or container.
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+To start collecting Spice runtime metrics:
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
Configure the Datadog Agent to scrape the Spice metrics endpoint:
```

File: spiceaidocs/docs/clients/Datadog/index.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+          - '*'
+```
+
+1. [Restart the Agent](https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent) to start collecting Spice runtime metrics.
+1. Refer to [Prometheus and OpenMetrics metrics collection from a host](https://docs.datadoghq.com/integrations/guide/prometheus-host-collection/) for all available configuration options and supported parameters.
+1. Open Datadog Metrics Explorer and type `spice` to confirm Spice telemetry information is successfully collected.
+
+<img width="800" src="/img/datadog/spice_datadog_metrics_explorer.png"/>
+
+## Import Spice.ai Dashboard
```
Comment (lukekim) [inline spiceaidocs/docs/clients/Datadog/index.md]:
```suggestion
## Import the Spice Datadog Dashboard
```

File: spiceaidocs/docs/clients/Datadog/index.md (hunk 1 above)
Comment (sgrebnov) [reply spiceaidocs/docs/clients/Datadog/index.md:5]: Updated, thank you

**Review by lukekim — APPROVED** [review]

**Pull request merged**